src/components/ProfileBlock.tsx

```diff
@@ -1,15 +1,113 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { Button } from "antd";
 import { useAppSelector } from "../redux/hooks";
+import FollowersFollowing from "./FollowersFollowing";
+import UserAvatar from "./UserAvatar";
 
 const ProfileBlock = (): JSX.Element => {
   const profile = useAppSelector((state) => state.user.profile);
+  const socialAddress = profile?.socialAddress;
+
+  const handle = "insert_handle_here";
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const [newName, setNewName] = useState<string | null>(null);
+  const [newHandle, setNewHandle] = useState<string | null>(null);
+  const [didEditProfile, setDidEditProfile] = useState<boolean>(false);
+
+  const profileName = profile?.name || "Anonymous";
+
+  useEffect(() => {
+    if (isEditing) {
+      if (
+        (newName !== null && newName !== profileName) ||
+        (newHandle !== null && newHandle !== handle)
+      ) {
+        setDidEditProfile(true);
+        return;
+      }
+      setDidEditProfile(false);
+    }
+  }, [isEditing, newName, newHandle, profileName]);
+
+  const getClassName = (sectionName: string) => {
+    return isEditing
+      ? `Profile__${sectionName} Profile__editing`
+      : `Profile__${sectionName}`;
+  };
+
+  const saveEditProfile = () => {
+    //this is where we write to the blockchain
+    setIsEditing(!isEditing);
+  };
+
+  const cancelEditProfile = () => {
+    setIsEditing(!isEditing);
+    setNewName(null);
+    setNewHandle(null);
+  };
 
   return (
     <div className="Profile__block">
-      <h1>Profile</h1>
-      <div>Wallet Address: {profile?.walletAddress}</div>
-      <div>Social Address: {profile?.socialAddress}</div>
-      <div>Profile Name: {profile?.name || "null"}</div>
+      {socialAddress ? (
+        <>
+          <div className="Profile__personalInfoBlock">
+            <div className="Profile__avatarBlock">
+              <UserAvatar profile={profile || null} avatarSize="large" />
+              {isEditing ? (
+                <>
+                  <Button
+                    className="Profile__editButton"
+                    onClick={() => saveEditProfile()}
+                    disabled={!didEditProfile}
+                  >
+                    SAVE
+                  </Button>
+                  <Button
+                    className="Profile__editButton"
+                    onClick={() => cancelEditProfile()}
+                  >
+                    CANCEL
+                  </Button>
+                </>
+              ) : (
+                <Button
+                  className="Profile__editButton"
+                  onClick={() => setIsEditing(!isEditing)}
+                >
+                  EDIT
+                </Button>
+              )}
+            </div>
+            <div className="Profile__personalInfo">
+              <label className="Profile__personalInfoLabel">NAME</label>
```

Comment:
You can use CSS to make these uppercase. That way screen readers will handle it better. (also you don't have to edit each one when later you need to change it to not all uppercase)